File: JUNGOL/Beginner/#5946 숫자 삼각형2/숫자 삼각형2.cpp
#include <iostream>

using namespace std;

int main() {
	int n; cin>>n;

	if(n<1 || n>50 || !(n%2)) cout<<"INPUT ERROR!\n";
	else{
		int cnt = n*2-1;
		for(int i=0; i<n; i++, cnt-=2){
			for(int j=0; j<i*2; j++) printf(" ");
			for(int j=0; j<cnt; j++) printf("%d ", i);
			cout<<endl;
		}
	}
}

File: JUNGOL/Beginner/#5946 숫자 삼각형2/숫자 삼각형2.py
def main():
    n = int(input())

    if n<1 or n>50 or n%2==0: print("INPUT ERROR!")
    else:
        cnt = n*2-1
        for i in range(n):
            print(' '*(i*2), end='')
            for j in range(cnt): print(i, end=' ')
            cnt -= 2
            print()

main()
